api/resources_portal/views/organization_grant.py

```diff
@@ -0,0 +1,68 @@
+from rest_framework import viewsets
+from rest_framework.exceptions import MethodNotAllowed
+from rest_framework.permissions import BasePermission, IsAuthenticated
+from rest_framework.response import Response
+
+from rest_framework_extensions.mixins import NestedViewSetMixin
+
+from resources_portal.models import Grant, GrantOrganizationAssociation, Organization
+from resources_portal.views.relation_serializers import GrantRelationSerializer
+
+
+class IsMemberOfOrganization(BasePermission):
+    def has_permission(self, request, view):
+        organization = Organization.objects.get(pk=view.kwargs["parent_lookup_organizations"])
+
+        return request.user in organization.members.all()
+
+
+class OwnsGrantAndOrganization(BasePermission):
+    def has_permission(self, request, view):
+        if view.action == "create":
+            grant = Grant.objects.get(pk=request.data["id"])
+            organization = Organization.objects.get(pk=view.kwargs["parent_lookup_organizations"])
+        else:
+            grant = Grant.objects.get(pk=view.kwargs["pk"])
+            organization = Organization.objects.get(pk=view.kwargs["parent_lookup_organizations"])
+
+        return request.user in grant.users.all() and request.user == organization.owner
```

Comment:
If the user isn't the owner of the org, we don't need to query the grants table at all. Can you move the check for the owner above the condition where we retrieve the grant and return False if they aren't? Something like this:
```python
organization = Organization.objects.get(pk=view.kwargs["parent_lookup_organizations"])

if not request.user == organization.owner:
    return False

if view.action == "create":
            grant = Grant.objects.get(pk=request.data["id"])
else:
...
```

Reply:
Good catch